workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java

```diff
@@ -164,6 +168,24 @@ public WorkReport execute(WorkFlowRequestDTO workFlowRequestDTO) {
 		return new DefaultWorkReport(WorkStatus.IN_PROGRESS, workContext, null);
 	}
 
+	/**
+	 * Merge the context of the previous execution into the top level of this current
+	 * context. Don't override in case of collisions.
+	 * @param executionId the execution context with the source context arguments
+	 * @param target target context to put arguments on
+	 */
+	private void mergeContextArgumentsFromExecution(UUID executionId, WorkContext target) {
+		WorkFlowExecution invokedBy = workFlowRepository.getById(executionId);
+		if (invokedBy != null) {
```

Comment:
I suggest wrapping this with `com.redhat.parodos.common.exceptions.ResourceNotFoundException` in case of failure to find it.
the global exception handler will know to handle it properly.